### PR TITLE
Remove all usages of GT_ItemStack2 as Set/Map keys by using fastutil

### PR DESCRIPTION
**Pull request by tth05**
This should be properly tested and reviewed in-case I broke something.

Memory allocations when starting GT5 dev environment:

_Before_
image

_After_
image

### CODE REVIEW

**Comment by bombcar** [comment]
in gt_mod.java update gtnhlib to minimum on the version that has fastutil:

`+ " required-after:gtnhlib@[0.0.8,);"` should be `+ " required-after:gtnhlib@[0.2.1,);"`

**Comment by chochem** [comment]
This PR seems to cause serious issues with NEI lookups. see in discord starting here https://discord.com/channels/181078474394566657/603348502637969419/1197220271464124467

File: src/main/java/gregtech/api/util/GT_OreDictUnificator.java [hunk 1]
```diff
@@ -41,9 +41,12 @@
 public class GT_OreDictUnificator {
 
     private static final Map<String, ItemStack> sName2StackMap = new HashMap<>();
-    private static final Map<GT_ItemStack2, ItemData> sItemStack2DataMap = new HashMap<>();
-    private static final Map<GT_ItemStack2, List<ItemStack>> sUnificationTable = new HashMap<>();
-    private static final Set<GT_ItemStack2> sNoUnificationList = new HashSet<>();
+    private static final Map<ItemStack, ItemData> sItemStack2DataMap = new Object2ObjectOpenCustomHashMap<>(
```
Comment (slprime) [inline src/main/java/gregtech/api/util/GT_OreDictUnificator.java:44]: properties `sUnificationTable` and `sItemStack2DataMap` use in `GregTech_API.sItemStackMappings`, and expected `GT_ItemStack` in https://github.com/Minecraft-TA/GT5-Unofficial/blob/6a02debedd95c56a5fcfd4bb46334239d64a0b63/src/main/java/gregtech/GT_Mod.java#L814

Comment (tth05) [reply src/main/java/gregtech/api/util/GT_OreDictUnificator.java:44]: I changed the line you linked as well

**Comment by tth05** [comment]
> This PR seems to cause serious issues with NEI lookups. see in discord starting here [discord.com/channels/181078474394566657/603348502637969419/1197220271464124467](https://discord.com/channels/181078474394566657/603348502637969419/1197220271464124467)

I think this should be fixed now

**Comment by slprime** [comment]
last commit fix nei

**Review by slprime — APPROVED** [review]

**Pull request merged**